google-cloud-examples/src/main/java/com/google/cloud/examples/storage/objects/StreamObjectDownload.java

```diff
@@ -0,0 +1,50 @@
+package com.google.cloud.examples.storage.objects;
+
+// [START storage_stream_file_download]
+
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+public class StreamObjectDownload {
+
+  public static void streamObjectDownload(
+      String projectId, String bucketName, String objectName, String targetFile)
+      throws IOException {
+    // The ID of your GCP project
+    // String projectId = "your-project-id";
+
+    // The ID of your GCS bucket
+    // String bucketName = "your-unique-bucket-name";
+
+    // The ID of your GCS object
+    // String objectName = "your-object-name";
+
+    // The path to the file to download the object to
+    // String targetFile = "path/to/your/file";
+
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    try (ReadChannel reader = storage.reader(BlobId.of(bucketName, objectName))) {
+      Path targetFilePath = Paths.get(targetFile);
+      FileChannel targetFileChannel = FileChannel.open(targetFilePath, StandardOpenOption.WRITE);
```

Comment:
One more improvement I missed last time.

We can include the FileChannel in the try-with-resource so that it's automatically closed along with the ReadChannel when leaving the block.

```suggestion
    Path targetFilePath = Paths.get(targetFile);

    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
    try (ReadChannel reader = storage.reader(BlobId.of(bucketName, objectName));
        FileChannel targetFileChannel =
            FileChannel.open(targetFilePath, StandardOpenOption.WRITE)) {

```